GoogleForCreators/web-stories-wp
# Plugin: Fix logic for plugin activation/deactivation

## Context

There are some issues with how we set set up / tear down things during activation/deactivation or when adding/removing sites.

## Summary

Following up from #7489.

This PR does following. 

- Removes all the useless `Activateable` / `Deactivateable` logic that is not being used / working. 
- Improves comments to make bootstrap cleaner
- Fixes a bug deleting sites where the wrong class was instantiated and the wrong method was called.
- Stops adding unnecessary rewrite rules
- Correctly removes rewrite rules upon plugin deactivation
- Correctly removes caps upon plugin deactivation
- Adds a new DB migration to flush rewrite rules (so unnecessary rewrites will be removed)


## Relevant Technical Choices

- Remove `Activateable` / `Deactivateable` usage

## To-do

- Automated test coverage

## User-facing changes

N/A

## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Verify the permalinks are correctly flushed, and capabilities are correctly set up upon plugin activation, and when adding a new site
2. Verify rewrite rules are correctly updated and capabilities are removed upon deactivation and site removal

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



See #7488
